src/log.rs

```diff
@@ -162,4 +165,76 @@ impl Writer {
 
         Ok(())
     }
+
+    pub fn compact_log<S: Serialize, P: AsRef<Path>>(
+        &self,
+        path: P,
+        data: Vec<S>,
+    ) -> Result<(), Error> {
+        let new_db_path = path.as_ref().with_file_name(Uuid::new_v4().to_string());
+
+        let mut new_db = OpenOptions::new()
+            .read(true)
+            .create(true)
+            .append(true)
+            .open(&new_db_path)
+            .map_err(|err| {
+                Error::OsError(
+                    format!(
+                        "While creating compaction file at {:?}, we received an error from the OS: {}",
+                        new_db_path.as_ref(),
+                        err
+                    ),
+                    err,
+                )
+            })?;
+
```

Comment:
don't duplicate anything like this, shape the functions that already do these things so that their logic can be re-used.